src/pages/index.js

```diff
@@ -23,45 +34,172 @@ import {
+    //likeBtn active if isLiked=true
+
+    // load user bio from server
+    const name = serverUserInfo.name;
+    const job = serverUserInfo.about;
+    const avatar = serverUserInfo.avatar;
+    // add user bio to DOM
+    userProfile.setUserInfo(name, job);
+    userProfile.setAvatar(avatar);
+  })
+  .catch((err) => {
+    console.error(err);
+  });
 // handler
+
 function handleAssignEditInput() {
   const userProfileData = userProfile.getUserInfo();
   editPopup.setInputValues(userProfileData);
 }
 
+function handleAvatarSubmit(inputObj) {
+  avatarEditPopup.showUploadingState();
+  const avatarLink = inputObj[avatarLinkInputId];
+  api
+    .updateProfileImage(avatarLink)
+    .then(avatarEditPopup.showUploaded())
+    .catch((err) => {
+      console.error(err);
+    });
+  avatarImg.src = avatarLink;
+  avatarEditPopup.close();
+  formValidators[avatarFormId].disableSubmitButton();
+  avatarEditPopup.reset();
+}
+
 function handleProfileSubmit(inputObj) {
-  userProfile.setUserInfo(
-    inputObj[editInputTitleId],
-    inputObj[editInputDescriptionId]
-  );
+  const nameEl = inputObj[editInputTitleId];
+  const jobEl = inputObj[editInputDescriptionId];
+  editPopup.showUploadingState();
+  api
+    .updateUserInfo({
+      editFormNameInput: nameEl,
+      editFormAboutInput: jobEl,
+    })
+    .then(editPopup.showUploaded())
+    .catch((err) => {
+      console.error(err);
+    });
+  // sent inputs to server
+  userProfile.setUserInfo(nameEl, jobEl);
+  // get userinfo from server and apply to profile
   editPopup.close();
+  formValidators[profileEditFormId].disableSubmitButton();
+  editPopup.reset();
 }
 function creatCard(item) {
-  const cardElement = new Card(item, cardTempSelector, handleImageClick);
+  const cardElement = new Card(
+    item,
+    cardTempSelector,
+    handleImageClick,
+    handleDeleteClick,
+    handleLikeClick
+  );
   const cardItem = cardElement.getView();
   return cardItem;
 }
+
 function handleAddCardSubmit(inputObj) {
   addCardPopup.close();
-  const newCard = creatCard({
-    name: inputObj[addCardInputTitleId],
-    link: inputObj[addCardInputLinkId],
-  });
-  cardList.addItem(newCard);
-  formValidators[addCardFormId].disableSubmitButton();
-  addCardPopup.reset();
+  const cardName = inputObj[addCardInputTitleId];
+  const cardLink = inputObj[addCardInputLinkId];
+  addCardPopup.showUploadingState();
+  api
+    .addNewCard({ cardElementName: cardName, cardElementLink: cardLink })
+    .then((card) => {
+      addCardPopup.showUploaded();
+      const newCard = creatCard(card);
+      cardList.addItem(newCard);
+      formValidators[addCardFormId].disableSubmitButton();
+      addCardPopup.reset();
+    })
+    .catch((err) => {
+      console.error(err);
+    });
 }
 
 function handleImageClick({ name, link }) {
   imageInfo.open({ name, link });
 }
+function handleLikeClick(card) {
+  const actionPromise = card.isLiked
+    ? api.unlikeCard(card).catch((err) => {
```

Comment:
Needs correction: please don't handle errors here, handle them on line 183 after handling the success